sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -73,6 +73,36 @@ private case class MsSqlServerDialect() extends JdbcDialect with NoLegacyJDBCErr
       }
     }
 
+    override def visitCaseWhen(children: Array[String]): String = {
+      // Since MsSqlServer cannot handle boolean expressions inside
+      // a CASE WHEN, it is necessary to convert those to an IIF
+      // expression that will return 1 or 0 depending on the result.
+      // Example:
+      // In:  ... CASE WHEN a = b THEN c = d ...
+      // Out: ... CASE WHEN a = b THEN IIF(c = d, 1, 0) ...
```

Comment:
how did you decide to use IIF over CASE WHEN? quick search indicates CASE WHEN is more portable

Reply:
IIF is more readable. This is only for the MsSql, other dialects will have their own implementations.

Reply:
reverted to CASE WHEN as I made a reusable function in a parent class